frontend/src/pages/Home/HomePage.vue

```diff
@@ -32,24 +31,19 @@ export default defineComponent({
     Loading,
   },
   setup() {
-    const fetchGems = inject(FETCH_GEMS_PENDING_COLLECTION, () =>
-      Promise.resolve([])
-    );
-    const gems = ref<Gem[]>([]);
-    // TODO: remove hardcoded userid
-    fetchGems(2)
-      .then((resp) => {
-        gems.value = resp;
-      })
-      // TODO: better error handling
-      .catch((err) => console.log(err, "Failed to fetch gems"));
+    const store = useStore();
 
-    return {
-      gems,
-    };
+    const fetchGems = () => store.dispatch("getGemsPendingCollectionForUser");
+    fetchGems();
+  },
+  computed: {
+    ...mapState({
+      gems: (state: any) => state.gems.gemsPendingCollection as Gem[],
```

Comment:
Any idea how to add types to the state? :'(

Reply:
The suggested solutions I managed to find online all seem to make the code much more unreadable. Guess we can just stick to this?